File: _test_OK/traci_example/TraCI-Matlab_py/MATLAB_TraCI/roundabout.m
%% SUMO TraCI example
clear all
close all
format compact %smaller line spacing in Command Window
clc %clears the Command Window

javaaddpath('traci4matlab.jar')

%% Main
projectPath = [pwd filesep 'roundabout.sumocfg'];

try
system(['sumo-gui' ' -c ' projectPath ' --remote-port 8813' ' --step-length    0.1' ' --start &']);
catch err
end

%initialization
[traciVersion,sumoVersion] = traci.init()

%sets the visualization scheme
traci.gui.setSchema('View #0', 'real world');

while i < 3600*10 %10 simulation steps (car following model) per second
%this runs one simulation step
traci.simulationStep();
%this is not necessary, only slows the simulation down for better display
pause(0.5)

%gets vehicle IDs
vehicles= traci.vehicle.getIDList();

%sets how the values set by setSpeed() and slowDown() shall be treated
    for ii=1:length(vehicles)
    traci.vehicle.setSpeedMode(cell2mat(vehicles(ii)),0);
    traci.vehicle.setSpeed(cell2mat(vehicles(ii)),55);
    %get actual speed, CO2 emission, distance travelled and edge ID of cars
    disp(['Speed ', cell2mat(vehicles(ii)), ': ', num2str(traci.vehicle.getSpeed(cell2mat(vehicles(ii)))) ' m/s']);
    disp(['CO2Emission ', cell2mat(vehicles(ii)), ': ', num2str(traci.vehicle.getCO2Emission(cell2mat(vehicles(ii)))) ' mg']);
    disp(['EdgeID of veh ', cell2mat(vehicles(ii)), ': ', num2str(traci.vehicle.getRoadID(cell2mat(vehicles(ii))))]);
    disp(['Distance1 ', cell2mat(vehicles(ii)), ': ', num2str(traci.vehicle.getDistance(cell2mat(vehicles(ii)))) ' m']);
    end

i=i+1;
end

traci.close();

File: _test_OK/traci_example/Python_TraCI/roundabout.py
import os, sys
import time

if 'SUMO_HOME' in os.environ:
    tools = os.path.join(os.environ['SUMO_HOME'], 'tools')
    sys.path.append(tools)
else:   
    sys.exit("please declare environment variable 'SUMO_HOME'")
    
import traci
import traci.constants

sumoCmd = ["sumo-gui", "-c", "roundabout.sumocfg", "--start"]
traci.start(sumoCmd)

print("Starting SUMO")
traci.gui.setSchema("View #0", "real world")
    
j = 0;

while(j<60):
    #this runs one simulation step
    time.sleep(0.5);
    traci.simulationStep();
    
    vehicles=traci.vehicle.getIDList();
    if (j%10)==0: #every 10 sec....

        for i in range(0,len(vehicles)): 
            #print(len(vehicles))
            print(vehicles[i])
            traci.vehicle.setSpeedMode(vehicles[i],0)
            #sets the speed of vehicles to 15 (m/s)
            traci.vehicle.setSpeed(vehicles[i],15)
            #get actual speed, emission, edge ID and total distance travelled of vehicles
            print("Speed ", vehicles[i], ": ",traci.vehicle.getSpeed(vehicles[i]), " m/s")
            print("CO2Emission ", vehicles[i], ": ", traci.vehicle.getCO2Emission(vehicles[i]), " mg/s")
            print("EdgeID of veh ", vehicles[i], ": ", traci.vehicle.getRoadID(vehicles[i]))
            print('Distance ', vehicles[i], ": ", traci.vehicle.getDistance(vehicles[i]), " m")
        
    j = j+1;
    
#get network parameters
IDsOfEdges=traci.edge.getIDList();
print("IDs of the edges:", IDsOfEdges)
IDsOfJunctions=traci.junction.getIDList();
print("IDs of junctions:", IDsOfJunctions)

traci.close()